src/hackerspace_online/tests/test_management_commands.py

```diff
@@ -128,3 +131,60 @@ def test_added_rows_to_db(self):
         self.assertEqual(Quest.objects.count(), expected_quest_count)
         self.assertEqual(Category.objects.count(), expected_campaign_count)
         self.assertEqual(User.objects.count(), expected_user_count)
+
+
+class FullCleanTest(TestCase, CommandMixin):
+    name = 'full_clean'
+
+    def setUp(self):
+        call_command('initdb')
+
+        # have to create tenant this way to prevent errors
+        # + "you cant create a tenant outside public schema"
+        with schema_context(get_public_schema_name()):
+            self.tenant1 = Tenant.objects.create(schema_name='test_schema1', name='Test Tenant 1')
+        with schema_context(get_public_schema_name()):
+            self.tenant2 = Tenant.objects.create(schema_name='test_schema2', name='Test Tenant 2')
+
+        # add two different errors to each tenant
+        with schema_context(self.tenant1.schema_name):
+            # will cause an error because author is None because of
+            # `null=True` without `blank=True`
+            #  ie. `{'author': ['this field cannot be blank.']}`
+            baker.make('announcements.Announcement')
+
+        with schema_context(self.tenant2.schema_name):
+            # will cause an error because semester is None because of
+            # `null=True` without `blank=True`
+            #  ie. `{'semester': ['this field cannot be blank.']}`
+            baker.make('quest_manager.QuestSubmission')
+
+    def test_full_clean(self):
+        """ Checks if full clean captures expected validation errors from "full_clean" management command
+        See setUp for the expected errors.
+        """
+        # capture stdout through contextlib.redirect_stdout, as the return value in call_command only works sometimes?
+
+        with StringIO() as buf, redirect_stdout(buf):
+            self.call_command(
+                '--tenants', 'test_schema1'
+            )
+
+            # profile is here because of grad_year being `None` for admin and owner
+            # grad_year is `null=True` and `blank=False`. Which means that objects with None will fail full clean.
+            # ie. `{'grad_year': ['this field cannot be blank.']}`
+            # This issue might be fixed once full clean is implemented for tenants app https://github.com/bytedeck/bytedeck/issues/1630
+            self.assertTrue('Profile' in buf.getvalue())
+
+            self.assertTrue('Announcement' in buf.getvalue())
```

Comment:
Why will this be True?

Reply:
added

```python
# should capture any print statements by self.call_command
# "Exception found on cleaning "<Object Name>" (<Model Name>) of type <Error Name>: <Error Log>"
 log = buf.getvalue()
```

Reply:
I meant specifically, on that line, why is Announcement failing the full_clean?

Reply:
in the setup where i initialized the announcement object i put:

```python
# will cause an error because author is None because of
# `null=True` without `blank=True`
#  ie. `{'author': ['this field cannot be blank.']}`
baker.make('announcements.Announcement')
```

Reply:
That explanation needs to be with the line of code, or at least in the test's docstring.

Reply:
added that comment to `self.assertTrue('Announcement' in log)`, and added a comment for `self.assertTrue('QuestSubmission' in log)`